### PR TITLE
Fixes for Build/Run Issues on Windows Platform

### PR DESCRIPTION
**Pull request by awgh**
These commits fix three separate issues in the extension when building or running on Windows:
1) The syntax in package.json (NODE_ENV=production node) does not work on Windows.  To be cross-platform, the cross-env Node package was added as a dependency and the build task modified to use it.  This should make builds work on Windows as well as other platforms.
2) Use of the child_process.spawn() call on Windows requires the shell: true option.  This has been added and now the VLS server will actually spawn on Windows.
3) Use of the child_process.execFile() call does not work on Windows since it does not spawn a shell either.  Use of this in the execV function has been replaced with child_process.exec(), which works fine because it does spawn a shell.

These patches address many open issues, including 230 and 243, but possibly many others.

### CODE REVIEW

**Comment by awgh** [comment]
After reviewing the open issues further, I believe this PR will fix 226, 228, 230, and 243 for a start.

**Comment by awgh** [comment]
Update:  I've added another fix for an issue detecting whether or not VLS was loaded on Windows, related to JSON.parse not being able to handle the backslashes in the path of VLS on Windows returned by "v ls".

File: src/langserver.ts [hunk 1]
```diff
@@ -40,8 +40,9 @@ export async function checkVlsInstallation(): Promise<boolean> {
 function receiveLauncherJsonData(cb: (d: { error?: { code: number, message: string }, message: string }) => void) {
 	return (rawData: string | Buffer) => {
 		const data = typeof rawData == 'string' ? rawData : rawData.toString('utf8');
-		log(`[v ls] new data: ${data}`);
-		cb(JSON.parse(data));
+		const escapedData = data.replaceAll('\\','/'); // replace backslashes found in Windows paths to prevent JSON parsing errors, TODO: proper JSON escaping solution needed
+		log(`[v ls] new data: ${data}\tescaped: ${escapedData}`);
+		cb(JSON.parse(escapedData));
```
Comment (nedpals) [inline src/langserver.ts:45]: would be great to have this in a separate PR so we can easily rollback changes in case something goes wrong

Comment (nedpals) [reply src/langserver.ts:45]: ok nevermind they are all connected lol

**Review by nedpals — CHANGES_REQUESTED** [review]
Just fix the code formatting errors and you're good. :+1:

**Review by nedpals — APPROVED** [review]
welp i'll just fix the formatting issues. thanks again!
